.circleci/config.yml

```diff
@@ -160,12 +148,102 @@ jobs:
           path: workspace/test-results
       - save_cache:
           name: Save Rust cache
-          key: rust-v1-{{ .Environment.CACHE_VERSION }}-{{ .Branch }}-{{ checksum "Cargo.lock" }}
+          key: rust-v1-integration-test-{{ .Environment.CACHE_VERSION }}-{{ .Branch }}-{{ checksum "Cargo.lock" }}
           paths:
             - target
             - ~/.cargo/registry
             - ~/.cargo/git
 
+  test-rust:
+    docker:
+      - image: python:3.12-slim-bookworm
+        auth:
+          username: $DOCKER_USER
+          password: $DOCKER_PASS
+        environment:
+          RUST_BACKTRACE: 1
+          RUST_TEST_THREADS: 1
+      - image: google/cloud-sdk:latest
+        auth:
+          username: $DOCKER_USER
+          password: $DOCKER_PASS
+        command: gcloud beta emulators bigtable start --host-port=localhost:8086
+    resource_class: xlarge
+    environment:
+      BIGTABLE_EMULATOR_HOST: localhost:8086
+    steps:
+      - checkout
+      # Need to download the poetry.lock files so we can use their
+      # checksums in restore_cache.
+      - restore_cache:
+          name: Restoring Rust cache
+          key: rust-v2-{{ .Environment.CACHE_VERSION }}-{{ .Branch }}-{{ checksum "Cargo.lock" }}
+      - run:
+          name: Create Workspace
+          command: mkdir -p workspace/test-results
+      - setup-rust
+      - run:
+          name: Set up cbt
+          command: |
+            echo "deb [signed-by=/usr/share/keyrings/cloud.google.gpg] https://packages.cloud.google.com/apt cloud-sdk main" | tee -a /etc/apt/sources.list.d/google-cloud-sdk.list
+            curl https://packages.cloud.google.com/apt/doc/apt-key.gpg | gpg --dearmor -o /usr/share/keyrings/cloud.google.gpg
+            apt-get update -y
+            apt install google-cloud-cli-cbt -y
+      - run:
+          name: Set up Rust
+          command: |
+            rustc --version
+      - run:
+          name: Setup Bigtable
+          command: scripts/setup_bt.sh
+      - run:
+          name: Install cargo-llvm-cov
+          command: cargo install cargo-llvm-cov
+      - run:
+          name: Report tests and coverage
+          command: |
+            cargo llvm-cov --json --output-path workspace/test-results/cov.json test --features=emulator --features=bigtable --jobs=2
```

Comment:
The following comment was previously present around this step, and i wonder if it should be preserved?

```
# Note: This build can potentially exceed the amount of memory availble to the CircleCI instance.
# We've seen that limiting the number of jobs helps reduce the frequency of this. (Note that
# when doing discovery, we found that the docker image `meminfo` and `cpuinfo` often report
# the machine level memory and CPU which are far higher than the memory allocated to the docker
# instance. This may be causing rust to be overly greedy triggering the VM to OOM the process.)
```